### PR TITLE
Competition_api class refactor

### PR DESCRIPTION
**Pull request by dunkOnIT**
Making competition api data more easy to work with

### CODE REVIEW

**Comment by dunkOnIT** [comment]
- [ ] move base_comp_url to `comeptition_api_url` in an .env file

**Comment by FinnIckler** [comment]
Maybe switch CompetitionAPI to something more rails-y like this

```
def find(competition_id)
    competition_json = fetch_competition(competition_id)
    if self.competition_exists?
      return CompetitionInfo.new(competition_json)
    end
    nil
  end
  
  def find!(competition_id)
    competition_json = fetch_competition(competition_id)
    if self.competition_exists?
      return CompetitionInfo.new(competition_json)
    end
    raise @error
  end
```

**Review by FinnIckler — CHANGES_REQUESTED** [review]
as per call

File: app/controllers/registration_controller.rb [hunk 1]
```diff
@@ -344,20 +334,20 @@ def is_admin_or_current_user?
...
 
-      competitor_limit = @competition[:competition_info]["competitor_limit"]
-      raise RegistrationError.new(:forbidden, ErrorCodes::COMPETITOR_LIMIT_REACHED) if params["competing"][:status] == 'accepted' && Registration.count > competitor_limit
+      raise RegistrationError.new(:forbidden, ErrorCodes::COMPETITOR_LIMIT_REACHED) if params['competing'][:status] == 'accepted' && Registration.count > @competition.competitor_limit
     end
 
     def get_competition_info!
-      if CompetitionApi.competition_exists?(@competition_id)
-        CompetitionApi.get_competition_info(@competition_id)
+      competition_response = CompetitionApi.new(@competition_id)
```
Comment (FinnIckler) [inline app/controllers/registration_controller.rb]: Didn't we say we change this to not be an initialized class?

File: app/controllers/registration_controller.rb [hunk 2]
```diff
@@ -286,50 +282,43 @@ def validate_guests!
     end
 
     def guests_valid?
-      @competition[:competition_info]["guest_entry_status"] != "restricted" || @competition[:competition_info]["guests_per_registration_limit"] >= params[:guests]
+      @competition.guest_entry_status != 'restricted' || @competition.guest_limit >= params[:guests]
```
Comment (FinnIckler) [inline app/controllers/registration_controller.rb:285]: I think it makes sense to have this as a guest_allowed method in the competition classe

**Review by FinnIckler — APPROVED** [review]
just one nit

File: app/controllers/registration_controller.rb [hunk 3]
```diff
@@ -344,20 +333,11 @@ def is_admin_or_current_user?
...
-      raise RegistrationError.new(:forbidden, ErrorCodes::COMPETITOR_LIMIT_REACHED) if params["competing"][:status] == 'accepted' && Registration.count > competitor_limit
-    end
-
-    def get_competition_info!
-      if CompetitionApi.competition_exists?(@competition_id)
-        CompetitionApi.get_competition_info(@competition_id)
-      else
-        raise RegistrationError.new(:not_found, ErrorCodes::COMPETITION_NOT_FOUND)
-      end
+      raise RegistrationError.new(:forbidden, ErrorCodes::COMPETITOR_LIMIT_REACHED) if params['competing'][:status] == 'accepted' && Registration.count > @competition.competitor_limit
```
[the remark this answers is not in the document]
Comment (dunkOnIT) [reply app/controllers/registration_controller.rb:341]: Got #258 open to track it :D

File: app/controllers/registration_controller.rb (hunk 2 above)
Comment (dunkOnIT) [reply app/controllers/registration_controller.rb]: Good point! Have changed it to `guest_limit_exceeded`, as that flowed better with the logic of checking for the `:guests` key first

**Pull request merged**